symengine/functions.cpp

```diff
@@ -3571,4 +3620,23 @@ RCP<const Basic> min(const vec_basic &arg)
     }
 }
 
+RCP<const Basic> expand_as_exp(const RCP<const Basic> &self)
+{
+    map_basic_basic mp_args;
+    for (const auto &a : self->get_args()) {
+        mp_args[a] = expand_as_exp(a);
+    }
+
+    if (is_a<Sin>(*self) or is_a<Cos>(*self) or is_a<Tan>(*self)
```

Comment:
Can we have two children classes of `TrigFunction`, which have trigonometric functions and their inverses into two separate classes.

Or just have a a special derived class from `TrigFunction` only for the inverse functions, and you can to `is_base_of<Trig> and not is_base_of<InvTrig>`. Upto you.